src/Common/tests/gtest_dns_reverse_resolve.cpp

```diff
@@ -0,0 +1,57 @@
+#include <gtest/gtest.h>
+#include <thread>
+#include <Common/DNSPTRResolverProvider.h>
+#include <Common/DNSResolver.h>
+#include <Poco/Net/IPAddress.h>
+#include <random>
+
+namespace DB
+{
+TEST(Common, ReverseDNS)
+{
+    auto addresses = std::vector<std::string>({
+        "8.8.8.8", "2001:4860:4860::8888", // dns.google
+        "142.250.219.35", // google.com
+        "157.240.12.35", // facebook
+        "208.84.244.116", "2600:1419:c400::214:c410", //www.terra.com.br,
+        "127.0.0.1", "::1"
+    });
+
+    // Good random seed, good engine
+    auto rnd1 = std::mt19937(std::random_device{}());
+
+    auto func = [&]()
+    {
+        for (int i = 0; i < 50; ++i)
+        {
+            auto & dns_resolver_instance = DNSResolver::instance();
+            dns_resolver_instance.setDisableCacheFlag();
+
+            auto addr_index = rnd1() % addresses.size();
+
+            auto result = dns_resolver_instance.reverseResolve(Poco::Net::IPAddress{ addresses[addr_index] });
+
+            if (result.empty())
+            {
+                std::cout<<"failed\n";
+            }
```

Comment:
Not sure what to assert for here as DNS requests might actually fail or one of the IP addresses actually become invalid in the future. Maybe it's enough to know it didn't crash and reached the end?

Just a note: this did not reproduce the issue, but it's good to have it.